app/services/org/create_created_plan_service.rb

```diff
@@ -6,24 +6,54 @@ def call(org = nil)
 
         orgs.each do |org|
           OrgDateRangeable.split_months_from_creation(org) do |start_date, end_date|
-            create_count_for_date(start_date: start_date, end_date: end_date, org: org)  
+            create_or_update(start_date: start_date, end_date: end_date, org: org)
           end
         end
       end
 
       private
 
+      def users(org)
+        User.where('users.org_id = ?', org.id)
+      end
+
+      def plans(start_date:, end_date: )
+        Plan.where('plans.created_at >= ? AND plans.created_at <= ?', start_date, end_date)
+      end
+
+      def creator_admon
+        Role.with_access_flags(:creator, :administrator)
+      end
+
       def count_plans(start_date: , end_date: , org:)
-        users = User.where('users.org_id = ?', org.id)
-        plans = Plan.where('plans.created_at >= ? AND plans.created_at <= ?', start_date, end_date)
-        creator_admon = Role.with_access_flags(:creator, :administrator)
+        users = users(org)
+        plans = plans(start_date: start_date, end_date: end_date)
```

Comment:
why set the results of the functions to variables of the same name here? Also prefer to use pluck here since you don't need the overhead of Rails creating the model. Do something like this:
```ruby
Role.joins([:plan, :user)
       .merge(creator_admin)
      .merge(users(org))
      .merge(plans(start_date: start_date, end_date: end_date))
      .pluck(:plan_id)
      .distinct.count
```

Reply:
Setting to variables improves the readability and using either (select or pluck) will not change much since active record queries are lazy evaluated and at the end just the count is returned, i.e. no models are instantiated.